src/plugins/mergeCells/mergeCells.js

```diff
@@ -647,16 +647,22 @@ class MergeCells extends BasePlugin {
    *
    * @private
    * @param {number} row Row index.
-   * @param {number} column Column index.
+   * @param {number} column Rendered column index.
```

Comment:
IMO should be visible. The `toVisualFromRenderableIndex` should be moved to `core.js` and translated in Walkontable callback before running `modifyGetCellCoords` hook.

Then te returned values can be translated back to the renderable indexes. Right now each hook needs to do it's own translation and back. There could be more `modifyGetCellCoords` hooks registered. And all our public API should use visual indexes whenever possible. WDYT?

Reply:
I've thought at first that it will be problematic. But, the `Walkontable` use `this.wot.getSetting('onModifyGetCellCoords', row, column)` function internally, it's not calling directly the hook.

Indexes translation back to renderable indexes is still needed and **this won't change**. The result of the hook is used by the `Walkontable`.

Thus, just removing one translation, modifying the `tableView` and it should work. Changed within: https://github.com/handsontable/handsontable/pull/6547/commits/a4d02be2e0068b629c4d6058f2b547cb4914abe7.

Reply:
@wojciechczerniak Btw. We have a function that should base on renderable indexes. It changes rendered indexes **by the reference** and the result is used only in Walkontalbe.

https://github.com/handsontable/handsontable/blob/a4d02be2e0068b629c4d6058f2b547cb4914abe7/src/plugins/mergeCells/mergeCells.js#L820-L842

Please look at the below line:

https://github.com/handsontable/handsontable/blob/a4d02be2e0068b629c4d6058f2b547cb4914abe7/src/plugins/mergeCells/mergeCells.js#L834